docs/styleguide/PyGuide.md

```diff
@@ -794,3 +798,87 @@ This is required so that the autogenerated API documentation is rendered properl
 
 If the autogenerated API documentation does not show type hints for certain arguments despite the fact that the type hints are present in the object's implementation code,
 or if the type hints do not refer to the API symbol's canonical alias, then the type hint should be explicitly declared in the docstring using the `:type *param_name*:` directive (or `:rtype:` for return types).
+
+<a id="s6-test-suite-coding-rules"></a>
+<a id="6-test-suite-coding-rules"></a>
+<a id="test-suite-coding-rules"></a>
+
+## 6 Test suite coding rules
+
+Unit tests are written and executed using the [pytest](https://pytest.org) framework.
+
+Parameters of test cases and fixtures should be type-hinted just like any other function.
```

Comment:
Split rules by headers like 
```
### Fixture
...
```

And will be good to add rule that the module versions should be determined in the test requirements.
```
torch  # bad 
torch == 2.1.0  # good
```

Reply:
Done

Reply:
Done what? Last commit is https://github.com/openvinotoolkit/nncf/pull/2220/commits/c84451794c24b64b2d86280fde629af16832fb33

Reply:
Pushed the new commit